net/lora-gateway-hal/Makefile

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright (C) 2019 Xue Liu <liuxuenetmail@gmail>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=lora-gateway-hal
+PKG_VERSION:=5.0.1
+PKG_RELEASE:=1
+
+PKG_SOURCE_URL:=https://codeload.github.com/Lora-net/lora_gateway/tar.gz/v$(PKG_VERSION)?
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_HASH:=1a0447d5e8183d08e6dce5f739f6872b9c57824b98f4078830d5ee21b15782c1
+PKG_MAINTAINER:=Xue Liu <liuxuenetmail@gmail.com>
+PKG_LICENSE_FILES:=LICENSE
+PKG_BUILD_DIR:=$(BUILD_DIR)/lora_gateway-$(PKG_VERSION)
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/libloragw
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE:=Driver/HAL library for Semtech SX1301
+  URL:=http://www.semtech.com/wireless-rf/lora.html
+endef
+
+define Package/libloragw/description
+  Driver/HAL library for Semtech SX1301 multi-channel modem and
+  SX1257/SX1255 RF transceivers.
+endef
+
+define Package/libloragw/config
+	source "$(SOURCE)/Config.in"
+endef
+
+define Package/libloragw-tests
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=LoRaWAN
+  TITLE:=Test programs for libloragw to check functionality
+  DEPENDS:=@libloragw
```

Comment:
Replace ```@``` with ```+```

Reply:
Hello @diizzyy 
Replacing `@` with `+` will resulte in error in `package/install` phase, becauese `package/libloragw/install` is not defined.
`package/libloragw/install` is not defined because `libloragw` is a static library. It is not necessary to install in device.
Sorry that. I don't know how to handle this.

Reply:
I'm actually facing similar problems with #8264 since even when I define `lora-picogw-hal` as a dependency of `lora-picogw-packet-forwarder`, the former gets removed [during the CI build](https://circleci.com/gh/openwrt/packages/1831):

```
find /home/build/build_dir/build_dir/target-mips_24kc_musl/picoGW_hal-0.2.2/ipkg-mips_24kc/lora-picogw-hal -name 'CVS' -o -name '.svn' -o -name '.#*' -o -name '*~'| xargs -r rm -rf
```

Which leads to `lora-picogw-packet-forwarder` not being able to link/compile properly:

```
make[4]: Entering directory '/home/build/build_dir/build_dir/target-mips_24kc_musl/picoGW_packet_forwarder-0.1.0/lora_pkt_fwd'
Makefile:17: ../../picoGW_hal-0.2.2/libloragw/library.cfg: No such file or directory
make[4]: *** No rule to make target '../../picoGW_hal-0.2.2/libloragw/library.cfg'.  Stop.
make[4]: Leaving directory '/home/build/build_dir/build_dir/target-mips_24kc_musl/picoGW_packet_forwarder-0.1.0/lora_pkt_fwd'
Makefile:11: recipe for target 'all' failed
make[3]: *** [all] Error 2
```

Maybe that CMake patch would make a lot of sense to apply against the picoGW* repos too, @xueliu? Alternatively, @diizzyy, is there a way to tell the pkg manager to compile the dep and leave it around for others to use *but* not include it on the final target system?

Thanks in advance to both!

Reply:
> I'm actually facing similar problems with #8264 since even when I define `lora-picogw-hal` as a dependency of `lora-picogw-packet-forwarder`, the former gets removed [during the CI build](https://circleci.com/gh/openwrt/packages/1831):
> 
> ```
> find /home/build/build_dir/build_dir/target-mips_24kc_musl/picoGW_hal-0.2.2/ipkg-mips_24kc/lora-picogw-hal -name 'CVS' -o -name '.svn' -o -name '.#*' -o -name '*~'| xargs -r rm -rf
> ```
> Which leads to `lora-picogw-packet-forwarder` not being able to link/compile properly:
> 
> ```
> make[4]: Entering directory '/home/build/build_dir/build_dir/target-mips_24kc_musl/picoGW_packet_forwarder-0.1.0/lora_pkt_fwd'
> Makefile:17: ../../picoGW_hal-0.2.2/libloragw/library.cfg: No such file or directory
> make[4]: *** No rule to make target '../../picoGW_hal-0.2.2/libloragw/library.cfg'.  Stop.
> make[4]: Leaving directory '/home/build/build_dir/build_dir/target-mips_24kc_musl/picoGW_packet_forwarder-0.1.0/lora_pkt_fwd'
> Makefile:11: recipe for target 'all' failed
> make[3]: *** [all] Error 2
> ```
I  think this compile error has nothing to do with dependence.

> Maybe that CMake patch would make a lot of sense to apply against the picoGW* repos too, @xueliu? Alternatively, @diizzyy, is there a way to tell the pkg manager to compile the dep and leave it around for others to use _but_ not include it on the final target system?
> 
I can add CMake for picoGW to let is compiled. But I have no Pico hardware to test.

> Thanks in advance to both!

Reply:
@xueliu Don't worry, I'll take care of the hardware testing side ;)

Reply:
@xueliu The compile error is due to `../../picoGW_hal-0.2.2/` disappearing under `lora-picogw-packet-forwarder`'s feet. How's that not related to dependence? `lora-picogw-hal` gets compiled first because it's a `DEPENDS`, but as soon as it gets built, it gets removed (`xargs -r rm -rf`) and then `lora-picogw-forwarder` cannot find it while it's compiling :/